modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/deployable/AbstractJsonDeserializingIncompleteArtifactModelResolver.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.deployable;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptor.MULE_ARTIFACT_JSON_DESCRIPTOR;
+
+import static java.lang.String.format;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.deployment.meta.AbstractMuleArtifactModelBuilder;
+import org.mule.runtime.api.deployment.meta.MuleDeployableModel;
+import org.mule.runtime.api.deployment.persistence.AbstractMuleArtifactModelJsonSerializer;
+import org.mule.runtime.core.api.util.IOUtils;
+import org.mule.runtime.module.artifact.activation.api.ArtifactActivationException;
+import org.mule.runtime.module.artifact.activation.api.deployable.ArtifactModelResolver;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDependency;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+/**
+ * Defines a template method for loading {@code deployableModels} for different kinds of deployable artifacts that might be
+ * incomplete,thus completing them with default values.
+ *
+ * @param <M> the concrete type of model to resolve.
+ * @param <B> the concrete type of builder to generate the complete model including default values.
+ */
+public abstract class AbstractJsonDeserializingIncompleteArtifactModelResolver<M extends MuleDeployableModel, B extends AbstractMuleArtifactModelBuilder<B, M>>
```

Comment:
Wouldn't it be better to remove the `final` qualifier from `JsonDeserializingArtifactModelResolver` so it can be used as parent here?

Reply:
yes, go ahead with that